python/spec/dependabot/python/update_checker/pipenv_version_resolver_spec.rb

```diff
@@ -387,6 +387,24 @@
         project_dependency_files("pipenv/missing-system-library")
       end
 
+      it "raises a helpful error" do
+        expect { subject }
+          .to raise_error(Dependabot::DependencyFileNotResolvable) do |error|
+            expect(error.message).to include(
+              "ERROR:pip.subprocessor:Getting requirements to build wheel exited with 1"
+            )
+          end
+      end
+    end
+
+    context "with a missing system libary, and when running python older than 3.12" do
+      # NOTE: Attempt to update an unrelated dependency (requests) to cause
+      # resolution to fail for rtree which has a system dependency on
+      # libspatialindex which isn't installed in dependabot-core's Dockerfile.
```

Comment:
This took me a second to understand, but I really appreciate the context you're providing here 👍

Reply:
I cannot say I understand it fully either 😅, I just copied the existing spec that started failing with Python 3.12, so that we still keep the coverage for Python 3.11.